DeDRM_plugin/ineptepub.py

```diff
@@ -446,49 +231,43 @@ def isPassHashBook(inpath):
                 return True
         except:
             pass
-        
+
     return False
 
-# Checks the license file and returns the UUID the book is licensed for. 
+# Checks the license file and returns the UUID the book is licensed for.
 # This is used so that the Calibre plugin can pick the correct decryption key
 # first try without having to loop through all possible keys.
-def adeptGetUserUUID(inpath): 
+def adeptGetUserUUID(inpath):
     with closing(ZipFile(open(inpath, 'rb'))) as inf:
         try:
             rights = etree.fromstring(inf.read('META-INF/rights.xml'))
             adept = lambda tag: '{%s}%s' % (NSMAP['adept'], tag)
             expr = './/%s' % (adept('user'),)
             user_uuid = ''.join(rights.findtext(expr))
-            if user_uuid[:9] != "urn:uuid:": 
+            if user_uuid[:9] != "urn:uuid:":
                 return None
             return user_uuid[9:]
         except:
             return None
 
-def verify_book_key(bookkey):
```

Comment:
Is there a particular reason you removed this function? Given that this plugin tries to use all available decryption keys, it's apparently possible that using a wrong key to decrypt the book key leads to garbage data, which this function is supposed to detect. Or is that redundant with your rewrite of how the decryption works (dropping OpenSSL support)?

See https://github.com/apprenticeharper/DeDRM_tools/issues/1862 for the reasons as to why this was added.

Reply:
> Or is that redundant with your rewrite of how the decryption works (dropping OpenSSL support)?

That is correct. The code that was using OpenSSL did not remove the padding, but the code that was (and is now) using PyCryptodome did. See: https://github.com/Legrandin/pycryptodome/blob/master/lib/Crypto/Cipher/PKCS1_v1_5.py . PyCryptodome removes the padding, hence the returned bookkey is always the proper length and doesn't start with `00 02`.

Two scenarios happen if the userkey or bookkey is incorrect: int(ciphertext) is >= than the modulus, in which case `ValueError` is raised; or the "sentinel" value is returned (I chose `None`).